apps/freenet-ping/types/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+use std::collections::HashSet;
+
+use chrono::{DateTime, Utc};
+
+pub use chrono;
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct Ping {
+    from: HashSet<String>,
+    timestamp: DateTime<Utc>,
```

Comment:
mmm no that's not what I meant...
each entry in the ping record is a tuple of `(time, string)`

If you want to keep using a hash map just do hashset<string, datetime> and then you remove all entries where the value is older than now...

in the contract you can use `stdlib/rust/src/time.rs` to get the now time from the host